src/blob_gc_job_test.cc

```diff
@@ -464,6 +464,58 @@ TEST_F(BlobGCJobTest, DeleteFilesInRange) {
   DestroyDB();
 }
 
+TEST_F(BlobGCJobTest, LevelMergeGC) {
```

Comment:
I think you need to disable purge_obsolete_files thread for the test, otherwise it can be flaky. How about change `StartBackgroundTasks` to not start purge_obsolete_files thread if `purge_obsolete_files_period_sec=0`, and set the option to 0 for this test?